source/hero/ControllableHero.hx

```diff
@@ -0,0 +1,49 @@
+package hero;
+
+import flixel.FlxG;
+import flixel.FlxSprite;
+import flixel.util.FlxColor;
+
+class ControllableHero extends FlxSprite
+{
+	public function new()
+	{
+		super(x, y);
+		makeGraphic(30, 30, FlxColor.WHITE, false, "ControllableHero");
+	}
+
+	override public function update(elapsed:Float)
+	{
+		if (FlxG.keys.pressed.LEFT)
+		{
+			velocity.x = -100;
+			velocity.y = 0;
+		}
+
+		if (FlxG.keys.pressed.RIGHT)
+		{
+			velocity.x = 100;
+			velocity.y = 0;
+		}
+
+		if (FlxG.keys.pressed.UP)
+		{
+			velocity.y = -100;
+			velocity.x = 0;
+		}
+
+		if (FlxG.keys.pressed.DOWN)
+		{
+			velocity.y = 100;
+			velocity.x = 0;
+		}
+
+		if (FlxG.keys.pressed.UP == false && FlxG.keys.pressed.DOWN == false && FlxG.keys.pressed.LEFT == false && FlxG.keys.pressed.RIGHT == false)
+		{
+			velocity.y = 0;
+			velocity.x = 0;
+		}
```

Comment:
This is functional, which is good, but it's also a bit heavy in terms of lines-of-code and conditional branching paths. 

A lot of game objects tend to get very busy `update()` methods (they've got a lot of stuff to do, after all) so in general I find it's a good idea to take blocks of self-contained logic like this and put them inside their own method. Then your `update()` method gets to look more like

```
override public function update(elapsed:Float)
{
    handleMovement(); // this is a function that handles all of the keyboard checking and sets velocity
    super.update();
}
```

Then if you want to add some new piece to `update()` you don't risk changing how your movement stuff works by accident.